eq/deflect/proxy.cpp

```diff
@@ -110,15 +99,29 @@ class Proxy::Impl : public boost::noncopyable
                                               ::deflect::BGRA, offsX, offsY );
         imageWrapper.compressionPolicy = ::deflect::COMPRESSION_ON;
         imageWrapper.compressionQuality = 100;
+        switch( channel.getEye( ))
+        {
+        case eq::EYE_LEFT:
+            imageWrapper.view = ::deflect::View::left_eye;
+            break;
+
+        case eq::EYE_RIGHT:
+            imageWrapper.view = ::deflect::View::right_eye;
+            break;
 
-        _sendFuture = _stream->asyncSend( imageWrapper );
+        default: break;
+        }
+
+        _pending[_current].emplace_back( _stream->asyncSend( imageWrapper ));
     }
 
     std::unique_ptr< ::deflect::Stream > _stream;
     std::unique_ptr< EventHandler > _eventHandler;
     Channel& _channel;
     lunchbox::Bufferb _buffer;
-    ::deflect::Stream::Future _sendFuture;
+    static const size_t _current = 0;
+    static const size_t _last = 1;
+    std::vector< ::deflect::Stream::Future > _pending[2];
```

Comment:
Now that there are two futures, shouldn't there be two _buffer as well? It seems to me that the current buffer could be overwritten before the last send has finished, are you sure this is not the case? Maybe even 4 buffers (2x each eye)?